src/gt4py/next/ffront/decorator.py

```diff
@@ -87,6 +87,11 @@ class Program:
         Sequence[str] | None
     )  # if the user requests static params, they will be used later to initialize CompiledPrograms
 
+    _offset_provider_extended_cache: eve_utils.CustomMapping = dataclasses.field(
+        default_factory=lambda: eve_utils.CustomMapping(common.offset_provider_hash),
+        kw_only=True,
+    )  # cache the extended dictionary of offset providers that includes the implicitly defined ones
```

Comment:
Just a suggestion, but I think this way is more clear that this is just an implementation detail.
```suggestion
    _offset_provider_extended_cache: eve_utils.CustomMapping = dataclasses.field(
        default_factory=lambda: eve_utils.CustomMapping(common.offset_provider_hash),
        kw_only=True,
        init=False
    )  # cache the extended dictionary of offset providers that includes the implicitly defined ones
```

Reply:
Good idea!